City-of-Helsinki/drupal-helfi-platform-config
# UHF-2158 District search

How to test:
* This is primarily meant to be used with the [drupal-helfi-sote](https://github.com/City-of-Helsinki/drupal-helfi-sote) site (but can also be used with other sites). Set it up first.
* Get the changes: `composer require drupal/helfi_platform_config:dev-UHF-2158-district-search`.
* Run `make shell`, and then:
  * Fetch some TPR Units: `MIGRATE_LIMIT=50 drush migrate:import tpr_unit`
  * Enable the new modules **helfi_tpr_unit_districts** and **helfi_helsinki_neighbourhoods**. The latter one will just create the Helsinki neighbourhood terms.
    * `drush en -y helfi_tpr_unit_districts helfi_helsinki_neighbourhoods`
  *  Clear caches: `drush cr`.

* Publish some of the imported TPR Units (in Finnish) and also their translations with some other language (e.g. in English).
* Create a landing page with Unit search paragraph. 
  * Add some or all of the published TPR Units to the Unit search and save the node.
  * Also translate the landing page and add some or all of the published TPR Unit translations to that page's Unit search.
* At this point the Unit search should work as before.
* Edit some of the TPR Units added to the Unit search:
  * Add one or many Helsinki neighbourhoods to the **Districts** field. The translated units are automatically updated with the same neighbourhoods.
* Now, when viewing the landing page, you should be able to search TPR Units using a neighbourhood as search word.

Especially check the following:
  * The search results do not have duplicate TPR Units.
  * The translated landing page is also working as expected.
  * If you go to the neighbourhoods taxonomy and modify term's translation, the changed translation is correctly shown at the corresponding TPR Unit translation. Also, at the translated landing page's Unit search, you can search using the new translation.

Also check the related PR [drupal-helfi-sote/pull/73](https://github.com/City-of-Helsinki/drupal-helfi-sote/pull/73).
